src/spikeinterface/core/generate.py

```diff
@@ -667,8 +686,7 @@ def synthesize_poisson_spike_vector(
 
     rng = np.random.default_rng(seed=seed)
 
-    if np.isscalar(firing_rates):
-        firing_rates = np.full(num_units, firing_rates, dtype="float64")
+    firing_rates = _ensure_firing_rates(firing_rates, num_units, seed)
```

Comment:
You can pass the rng as well as mentioned above?

Reply:
The thing is that if the rng is used before or not depending other global option then we do not have the same.
With this seed  given we somehow reset the generator whatever happen.

Reply:
You are correct. This is a very good point for reproducibility. Passing the rng means you are passing state which is wrong.